### PR TITLE
Only build pkg/sdn/plugin on linux

### PR DESCRIPTION
**Pull request by danwinship**
We are currently building (most of) the SDN code on Darwin and Windows, even though it is totally linux-specific.

I ran into this when trying to make more use of the "netlink" library in the sdn code (#15834). The library only builds on linux, so any file that imports it has to also be only built on linux.

So this tries to fix the main openshift binary to only include the SDN code when building on Linux. I don't know if it's correct stylistically. I'm happy to rewrite it however.

@openshift/networking FYI

### CODE REVIEW

**Comment by smarterclayton** [comment]
Yay, build-cross got triggered automatically.  So happy

**Comment by danwinship** [comment]
OK, cross-build succeeded. Other failures look like flakes.

I had to tweak hack/import-restrictions.json to get "make verify" to pass, and probably that should be fixed differently. I guess one approach would be to move the SDNNode and SDNProxy interfaces into a new directory (neither pkg/sdn/plugin nor pkg/sdn/apis) so we can keep pkg/sdn/apis pure?

File: hack/import-restrictions.json [hunk 1]
```diff
@@ -180,6 +180,7 @@
     ],
     "allowedImportPackageRoots": [
       "vendor/k8s.io/apimachinery",
+      "vendor/k8s.io/kubernetes/pkg/proxy/config",
```
Comment (smarterclayton) [inline hack/import-restrictions.json]: I think this won't be allowed - @deads2k for his goals here

Comment (deads2k) [inline hack/import-restrictions.json]: > I think this won't be allowed - @deads2k for his goals here

Our api packages should only be for the type definitions themselves.  We're going to build a client-go library for openshift (see attempts here: https://github.com/openshift/client-go/pull/2) and the dependencies for that library need to be reasonable.  Our API types are required, but the interfaces which make them more useable for runtime purposes aren't generally needed by the client, so they should live a separate package.

**Comment by danwinship** [comment]
/retest
after revert of bad commit to master

**Comment by danwinship** [comment]
flake #14385
/test extended_networking_minimal
/test extended_networking

**Comment by danwinship** [comment]
All tests pass now
@smarterclayton ok now? I moved the OpenShift-internal SDN interfaces to the top level of pkg/sdn to get rid of the extra imports in pkg/sdn/apis/network

**Comment by deads2k** [comment]
Glad to see the move to start slimming down the package

/approve

lgtm if it's basically a straight move

**Pull request merged**